error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Bubble`
  --> $DIR/try-operator-on-main.rs:9:5
   |
LL |     std::fs::File::open("foo")?; //~ ERROR the `?` operator can only
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Bubble` is not implemented for `()`
   = note: required by `std::ops::Bubble::bubble`

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Bubble`
  --> $DIR/try-operator-on-main.rs:9:5
   |
LL |     std::fs::File::open("foo")?; //~ ERROR the `?` operator can only
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the `?` operator cannot be applied to type `std::result::Result<std::fs::File, std::io::Error>`
   |
   = help: the trait `std::ops::Bubble<()>` is not implemented for `std::result::Result<std::fs::File, std::io::Error>`
   = note: required by `std::ops::Bubble::bubble`

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Bubble`
  --> $DIR/try-operator-on-main.rs:13:5
   |
LL |     ()?; //~ ERROR the `?` operator can only
   |     ^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Bubble<_>` is not implemented for `()`
   = note: required by `std::ops::Bubble::bubble`

error[E0277]: the trait bound `(): std::ops::Try` is not satisfied
  --> $DIR/try-operator-on-main.rs:16:5
   |
LL |     try_trait_generic::<()>(); //~ ERROR the trait bound
   |     ^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::ops::Try` is not implemented for `()`
   |
note: required by `try_trait_generic`
  --> $DIR/try-operator-on-main.rs:21:1
   |
LL | fn try_trait_generic<T: Try>() -> T {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Bubble`
  --> $DIR/try-operator-on-main.rs:23:5
   |
LL |     ()?; //~ ERROR the `?` operator can only
   |     ^^^ the `?` operator cannot be applied to type `()`
   |
   = help: the trait `std::ops::Bubble<_>` is not implemented for `()`
   = note: required by `std::ops::Bubble::bubble`

error[E0277]: the `?` operator can only be applied to values that implement `std::ops::Bubble`
  --> $DIR/try-operator-on-main.rs:23:5
   |
LL |     ()?; //~ ERROR the `?` operator can only
   |     ^^^ the `?` operator cannot be applied to type `T`
   |
   = help: the trait `std::ops::Bubble` is not implemented for `T`
   = help: consider adding a `where T: std::ops::Bubble` bound
   = note: required by `std::ops::Bubble::bubble`

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0277`.
