error[E0277]: the trait bound `MyErrorType: std::convert::From<&str>` is not satisfied
  --> $DIR/try-block-bad-type.rs:19:9
   |
LL |         Err("")?; //~ ERROR the trait bound `MyErrorType: std::convert::From<&str>` is not satisfied
   |         ^^^^^^^^ the trait `std::convert::From<&str>` is not implemented for `MyErrorType`
   |
   = note: required by `std::convert::From::from`

error[E0271]: type mismatch resolving `<std::result::Result<i32, i32> as std::ops::Try>::Ok == &str`
  --> $DIR/try-block-bad-type.rs:24:9
   |
LL |         "" //~ ERROR type mismatch
   |         ^^ expected i32, found &str
   |
   = note: expected type `i32`
              found type `&str`

error[E0271]: type mismatch resolving `<std::result::Result<i32, i32> as std::ops::Try>::Ok == ()`
  --> $DIR/try-block-bad-type.rs:27:39
   |
LL |     let res: Result<i32, i32> = try { }; //~ ERROR type mismatch
   |                                       ^ expected i32, found ()
   |
   = note: expected type `i32`
              found type `()`

error[E0277]: the trait bound `(): std::ops::Try` is not satisfied
  --> $DIR/try-block-bad-type.rs:29:23
   |
LL |     let res: () = try { }; //~ the trait bound `(): std::ops::Try` is not satisfied
   |                       ^^^ the trait `std::ops::Try` is not implemented for `()`
   |
   = note: required by `std::ops::Try::from_ok`

error[E0277]: the trait bound `i32: std::ops::Try` is not satisfied
  --> $DIR/try-block-bad-type.rs:31:24
   |
LL |     let res: i32 = try { 5 }; //~ ERROR the trait bound `i32: std::ops::Try` is not satisfied
   |                        ^^^^^ the trait `std::ops::Try` is not implemented for `i32`
   |
   = note: required by `std::ops::Try::from_ok`

error: aborting due to 5 previous errors

Some errors occurred: E0271, E0277.
For more information about an error, try `rustc --explain E0271`.
